src/components/App/App.js

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+// import logo from '../../logo.svg';
+import './App.css';
+import Display from "../Display/Display";
+import ButtonPanel from "../ButtonPanel/ButtonPanel";
+import {calculate } from "../../logic/calculate";
+
+class App extends Component {
+    constructor(props){
+        super(props);
+        this.state = {
+            calculator: {
+                total: null,
+                next: null,
+                operation: null
+            }
+        }
+    }
+    handleClick = (buttonName) => {
+        const calculator = calculate(this.state.calculator, buttonName);
+        this.setState(calculator);
+    }
+    _renderValueToDisplay = () => {
```

Comment:
We are calling `this.state.calculator` multiple times here. It's better if we use object destructuring to make our subsequent calls shorter:

`const { next, total, operation } = this.state.calculator`

And you can, for instance just use `total` instead of calling `this.state.calculator.total`